lib/rule-tester/rule-tester.js

```diff
@@ -794,6 +798,29 @@ class RuleTester {
             }
         }
 
+        /**
+         * Check if this test case is a duplicate of one we have seen before.
+         * @param {Object} item test case object
+         * @param {Set<string>} seenTestCases set of serialized test cases we have seen so far (managed by this function)
+         * @returns {void}
+         * @private
+         */
+        function checkDuplicateTestCase(item, seenTestCases) {
+            if ((Array.isArray(item.options) && item.options.some(i => typeof i === "object")) || item.settings || item.languageOptions?.parser || item.languageOptions?.parserOptions || item.plugins) {
```

Comment:
there is a room to improve it, but I think it's fine to just check these properties 👍  - as it's described in the rfc.

Reply:
It's a start.

However, I would love it if I could replace this logic with a simple function call to see if a given object is serializable. Surprisingly, I haven't found a third-party library to do this. Instead, I could try to piece together such a function myself based on StackOverflow snippets etc and add unit tests for it. Thoughts?

It might be best to solve this the right way now (in a way that avoids a lot of false negatives) since changing this later could be a breaking change.

Reply:
IMHO, it's not worth adding a new dependency: rule tester is shipped with eslint, but most eslint users don't really use it. A simple implementation to cover 90%+ cases is good enough for me. 😄

Reply:
I went ahead and added a `isSerializable()` function which uses only the tiny `lodash.isplainobject` as a dependency. While the previous approach could have been sufficient as you mentioned, this new approach is a lot more robust (no false negatives) and the code is cleaner.